lib/puppet/functions/classified_with.rb

```diff
@@ -0,0 +1,9 @@
+Puppet::Functions.create_function(:classified_with) do
```

Comment:
Sadly there's no yak shaving emote. That being said I'll rename them.

These functions get around the lack of parse order safety in Puppet by querying the classification state of the compiler before compilation even starts. This relies on the fact that masters are almost always classified with `puppet_enterprise::profile::master` via the classifier, so combined with the `defined()` check this is about as good as we'll get without linking the compiler against libtachyon.